lib/workers/repository/process/write.ts

```diff
@@ -27,31 +27,28 @@ export async function writeUpdates(
     }
     return true;
   });
-  let prsRemaining = await getPrsRemaining(config, branches);
+  const prsRemaining = await getPrsRemaining(config, branches);
   logger.debug({ prsRemaining }, 'Calculated maximum PRs remaining this run');
+  setLimit('prsRemaining', prsRemaining);
   for (const branch of branches) {
     addMeta({ branch: branch.branchName });
-    const prLimitReached = prsRemaining <= 0;
-    const commitLimitReached = getLimitRemaining('prCommitsPerRunLimit') <= 0;
-    const res = await processBranch(branch, prLimitReached, commitLimitReached);
+    const res = await processBranch(branch);
     branch.res = res;
     if (res === 'automerged' && branch.automergeType !== 'pr-comment') {
       // Stop procesing other branches because base branch has been changed
       return res;
     }
-    let deductPrRemainingCount = 0;
     if (res === 'pr-created') {
-      deductPrRemainingCount = 1;
+      incrementLimit('prsRemaining');
     }
     // istanbul ignore if
     if (
       res === 'automerged' &&
       branch.automergeType === 'pr-comment' &&
       branch.requiredStatusChecks === null
     ) {
-      deductPrRemainingCount = 1;
+      incrementLimit('prsRemaining');
```

Comment:
In more aggressive refactoring, we can move this `incrementLimit` calls right after the `platform.createPr(...)` calls we're interested in. In fact, I would do it here, but not sure about this `automerge` branch).